code-monkey-ts2/tools/invoke_agent_tool.ts

```diff
@@ -4,12 +4,25 @@ import { instrument, currentSpan } from "../instrumentation";
 // TODO import { getLogger } from "../utils/logger";
 
 import { getServiceForAgent } from "../agents";
+import { AsyncLocalStorage } from "async_hooks";
 
 const schema = z.object({
     agent_name: z.string().describe("Name of the agent to invoke"),
     prompt: z.string().describe("Prompt to run with the agent"),
 });
 
+
+// a hack to clear the async local storage langchain uses to send events to the correct stream.
```

Comment:
Nit: Let's move this to a separate file?
Also, generally, any code that executes on import is likely coming back to haunt you at some point in time. Putting it in a function might be a better idea.